### ISSUE TITLE
Upgrade BoringSSL

### ISSUE THREAD

**Issue by jonasfj** [google/webcrypto.dart#111]
We need to upgrade BoringSSL, however, their build scripts have changed a bit. In particular how ASM files are added, so we might need to make some changes.

The logic is in:
https://github.com/google/webcrypto.dart/blob/master/tool/update-boringssl.py

We need to bump `BORINGSSL_REVISION`.

And then fix our build script generator logic, since `asm_outputs` parameter isn't give to [`WriteFiles`](https://github.com/google/webcrypto.dart/blob/master/tool/update-boringssl.py#L135) anymore.

I tried playing with this, but we need to do more than simple remove the `asm_outputs` parameter.

--------

There is some documentation in: https://boringssl.googlesource.com/boringssl/+/refs/heads/master/INCORPORATING.md

The rest is reading BoringSSL code and build system :rofl:

**Comment by leanhdaoit** [comment]
Hi @jonasfj

Thank you for your continuous efforts in maintaining the webcrypto project.

Recently, I encountered an issue when trying to run Flutter project with webcrypto: ^0.5.7 in dependencies with XCode 16. The error message:
`unsupported` option '-G' for target `'arm64-apple-ios14-simulator`

This appears to be related to an outdated version of BoringSSL, as discussed in [this gRPC issue](https://github.com/grpc/grpc/issues/36888). The problem has been resolved in gRPC with [this pull request](https://github.com/grpc/grpc/pull/37310/files#diff-9e2ee0b81f22e886d633270c332d5716a81521346b9bd4f537f54a9e309b2f04), which updates BoringSSL.

Could you consider applying the update to the BoringSSL version used in webcrypto? This would ensure compatibility with the latest XCode version and benefit developers facing similar issues.

Thank you for your time and support!

**Comment by bhushangitfull** [comment]
I would like to help on this bug.

**Comment by jonasfj** [comment]
I have plans to upgrade boringssl, but haven't had the time.

Last I tried it turned out that the python script had changed a lot. So
we'll need to change how we generate lists of sources... And possibly how
we build too.

In any case, yes, it's something we want to do.

@bhushangitfull, you are most welcome to help.
I'd suggest making a fork, and trying to fix:
https://github.com/google/webcrypto.dart/blob/master/tool/update-boringssl.py

You could also try to the steps manually, and we can find out how to
automatically roll it later.
But I do thing we want rolls to be scripted, we don't want this package
laggy behind boringssl.

If you get something even partially working, please share as a separate
branch. I'm currently on parental leave, but back next week and will
hopefully find time to help with this eventually.

tors. 16. jan. 2025, 04.11 skrev Buddhbhushan Kamble <
***@***.***>:

**Comment by bhushangitfull** [comment]
I will give my best

**Comment by bhushangitfull** [comment]
Hi @jonasfj, What do you think about using gith hub action workflow for making the rolling process automatic?

**Comment by bhushangitfull** [comment]
Rolling and bump for boringssl is done, can I do PR

**Comment by bhushangitfull** [comment]
Her is the branch `https://github.com/bhushangitfull/webcrypto.dart-upgradeBoringSSl-.git`

**Comment by jonasfj** [comment]
Absolutely, I'm hoping you made a script for rolling the source code in. Because it's pretty hard to actually review all the code.

So maybe we can focus on just the changes required to automatically rolling in a updated BoringSSL.

**Comment by bhushangitfull** [comment]
My name is Buddhbhushan , and I'm reaching out because I'm interested in contributing to Dart as part of Google Summer of Code 2025. I was excited to see that Dart has been selected as a participating organization this year.

 I am a Third Year B.Tech student. I've been following Dart's project ideas list and I am particularly interested in webcrypto.dart.

I have read the [Contributor Guide](https://google.github.io/gsocguides/student/writing-a-proposalfor), and I want more information for building my proposal. So what things I need to for proposal?

**Comment by harrshita123** [comment]
Hi, I’d like to work on this issue.

I went through the details and it looks like the main work is not just updating the BoringSSL revision, but also adjusting the update/build generation logic because the way assembly files are handled has changed upstream.

I’ll start by looking into the current script and BoringSSL’s latest build metadata, then try to make the update flow work cleanly and run the tests before opening a PR.

Please let me know if there’s any specific BoringSSL revision you’d prefer me to use.
